Clean up so it now compiles with `make`.

hackerb9 opened the pull request:
Fixes several bugs. Could use more cleanup, but this is a good start.

Traace committed:
Include <string.h>

hackerb9 committed:
Initial support for compiling from source.

hackerb9 committed:
Don't include binaries in standard source code. (Should be in releases).

hackerb9 committed:
Fix recursive make to be conditional on cd

hackerb9 committed:
Fix up recursive make properly use $(MAKE), -k, and -C

hackerb9 committed:
Fix 'make clean' to clean the right directory

hackerb9 committed:
make clean shouldn't return an error if .o files don't exist

hackerb9 committed:
make clean shouldn't return an error if .o files don't exist

hackerb9 committed:
Don't include binaries in source code

hackerb9 committed:
.h files should not be executable.

hackerb9 committed:
.c files should not be executable

hackerb9 committed:
Don't call binary binhack32 (or 64). Just binhack.

hackerb9 committed:
Don't include upx binary in source.

hackerb9 committed:
Use system upx if available. Ignore upx, if not.

hackerb9 committed:
Ignore upx in "all" target as well as "linux".

hackerb9 committed:
"linux" target shouldn't duplicate code from "all"

hackerb9 committed:
Add -Wno-narrowing to ignore C++ narrowing errors

This is probably a bad idea, but it gets this code to compile in
modern compilers. We should check if there's already a newer version
of this program which we can find elsewhere.

hackerb9 committed:
Ignore BLOCKSIZE in du if user set it

hackerb9 committed:
Upx is way to garrulous. Make it quieter with -qq.

hackerb9 committed:
Remove -Wno-narrowing ugliness. Fix C++ error by using UNSIGNED char array.

hackerb9 committed:
Use cleaner sed syntax. No need to ignore errors from mkdir -p.

hackerb9 committed:
Add back in removing the build directory upon 'make clean'

It gets created by 'make package'.

hackerb9 committed:
Install binaries recursively

hackerb9 committed:
Use leading dash syntax instead of "|| true"

hackerb9 committed:
Rename 'build' target to 'compile'

so it won't get confused by 'build' directory and think there's
nothing to do.

hackerb9 committed:
Replace gets() with fgets()

hackerb9 committed:
#include common.h to define fwrite_as_big()

hackerb9 committed:
Add #include unistd.h for chdir()

hackerb9 committed:
#include <stdio.h> for fgets

hackerb9 committed:
Remove vestigal "compile_res" target

hackerb9 committed:
Don't recompile if the source files haven't changed.

hackerb9 committed:
MacOS and Linux compiling are equivalent

hackerb9 committed:
Don't recompile if source hasn't changed.

hackerb9 committed:
Don't recompile if source code hasn't changed

hackerb9 committed:
Ignore binary files in git

Nold360 commented:
Wow.. that's quite a lot - thanks for your contribution, but i'll need some time to test this.. but i really want to merge this PR.

hackerb9 commented:
That'd be great if you could merge! 

One change I didn't make was adding `upx` to the debian/control file as "recommended". I don't quite understand the role of upx in binhack. On my machine it does nothing. Perhaps upx should simply be removed completely?

hackerb9 committed:
Use binhack, not binhack32

hackerb9 committed:
Use binhack, not binhack32

hackerb9 committed:
Don't wait for user to hit ENTER before exiting

hackerb9 committed:
Don't wait for user to hit ENTER before exiting

hackerb9 committed:
add 'make test' target to run the test script

hackerb9 committed:
Don't wait for user to hit ENTER before exiting

Also, just run 'make', not 'make linux'.
It's the same no matter what UNIX-like system you run.

hackerb9 committed:
Don't wait for user to hit ENTER before exiting

Also, use -x so user can see what the script is actually doing.

hackerb9 committed:
Make the build_linux and clean_linux scripts executable

I'm not actually sure why they exist since UNIX systems will just use
make, but perhaps they are helpful as a Rosetta stone for people
switching over from Microsoft Windows.

hackerb9 committed:
Fixed debian package build to include binaries

hackerb9 committed:
Create new man page for cdirip.

hackerb9 committed:
cdirip was written by Nold, right?

hackerb9 committed:
Created man page for burncdi

hackerb9 committed:
fix double quotes within double quotes

hackerb9 committed:
Use a search path for the IP.BIN files

Instead of being hardcoded to a single directory, search through
several to find the IP.BIN file.

    ~/.mksdiso/
    ~/.local/share/mksdiso/
    /usr/local/share/mksdiso/
    /usr/share/mksdiso/

hackerb9 committed:
Add fallback if no directory with IP.BIN exists

hackerb9 committed:
Typos

hackerb9 committed:
Do not edit mksdiso script using sed from Makefile

hackerb9 committed:
Created man page for mksdiso

hackerb9 committed:
typo

hackerb9 committed:
Fix RET value so it shows an error if any loop failed

Previously, it overwrote RET=$? every time, so it would only show an
error if the final call to $BURNBIN failed.

hackerb9 committed:
Make errors stand out more.

hackerb9 committed:
Add -s (speed) option, default to fast burning instead of 4x

Even crummy CD burners seem to be able to do full speed these days.
No need to default to something slow. (In fact, sometimes I've seen
errors only when the drive is at a slow speed.)

hackerb9 committed:
Reset to tao mode

hackerb9 committed:
Add -n (not really) option to run `wodim -dummy`

Stop making coasters while debugging burncdi. :-)

hackerb9 committed:
Fix up incorrect error handling when wodim failed

hackerb9 committed:
Default to detecting CD-RW device instead of /dev/cdrw

hackerb9 committed:
Ask user whether to continue after a bad burn

Why waste time (and possibly ruin a disk?)

hackerb9 committed:
Add -v (verbose flag)

hackerb9 committed:
use -v flag to also make wodim more verbose

hackerb9 committed:
Typo

hackerb9 committed:
Merge branch 'manpages'

hackerb9 committed:
Remove vestigal tracksize calculation

hackerb9 committed:
Merge from jozip's 0.6.3 version

hackerb9 committed:
Merge from jozip's 0.6.3 version

hackerb9 committed:
Merge from jozip's 0.6.3 version

hackerb9 committed:
Merge from jozip's 0.6.3 version

hackerb9 committed:
Merge from jozip's 0.6.3 version

hackerb9 committed:
Merge from jozip's 0.6.3 version

hackerb9 committed:
Merge jozip/cdirip 0.6.3

hackerb9 committed:
New files from jozip/cdirip 0.6.3
